core/src/main/java/com/linecorp/armeria/server/Server.java

```diff
@@ -162,6 +163,51 @@ public String defaultHostname() {
         }
     }
 
+    /**
+     * Returns the primary {@link ServerPort}'s port that this {@link Server} is listening to.
+     *
+     * @throws IllegalStateException if no available active ports
+     */
+    public int activeLocalPort() {
+        synchronized (activePorts) {
+            ServerPort serverPort = Iterables.getFirst(activePorts.values(), null);
+
+            if (serverPort == null) {
+                throw new IllegalStateException("No available active local port");
+            }
+
+            return serverPort.localAddress().getPort();
+        }
+    }
+
+    /**
+     * Returns {@link ServerPort}'s port which is mapped to the given {@code protocol}.
+     *
+     * @throws IllegalStateException if no available active ports for given {@code protocol}
+     */
+    public int activeLocalPort(SessionProtocol protocol) {
+        requireNonNull(protocol, "protocol");
+
+        synchronized (activePorts) {
+            Predicate<ServerPort> portPredicate = serverPort -> {
```

Comment:
Could be extracted into a `private static` method so it's reused for `activeLocalPort()` above.